lib/matobjnz.gd

```diff
@@ -12,14 +12,78 @@
 # in the range [0..n-1], but reduce after
 # arithmetic. This way avoid always wrapping all entries separately
 
+#############################################################################
+##
+#R  IsZmodnZVectorRep( <obj> )
+##
+##  <#GAPDoc Label="IsZmodnZVectorRep">
+##  <ManSection>
+##  <Filt Name="IsZmodnZVectorRep" Arg='obj' Type='Representation'/>
+##
+##  <Description>
+##  An object <A>obj</A> in <Ref Filt="IsZmodnZVectorRep"/> describes
+##  a vector object (see <Ref Filt="IsVectorObj"/>) with enries in a
+##  residue class ring of the ring of integers (see <Ref Func="ZmodnZ"/>).
+##  <A>obj</A> is internally represented as a positional object
+##  (see <Ref Filt="IsPositionalObjectRep"/>) which stores the base domain
+##  (see <Ref Attr="BaseDomain" Label="for a vector object"/>)
+##  at position <M>1</M> and a plain list of integers at position <M>2</M>.
+##  <P/>
+##  <Ref Filt="IsZmodnZVectorRep"/> implies <Ref Filt="IsCopyable"/>,
+##  thus matrix objects in this representation can be mutable.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
 DeclareRepresentation( "IsZmodnZVectorRep",
         IsVectorObj and IsPositionalObjectRep
+    and IsCopyable
     and IsNoImmediateMethodsObject
     and HasBaseDomain and HasOneOfBaseDomain and HasZeroOfBaseDomain,
     [] );
 
+
+#############################################################################
+##
+#R  IsZmodnZMatrixRep( <obj> )
+##
+##  <#GAPDoc Label="IsZmodnZMatrixRep">
+##  <ManSection>
+##  <Filt Name="IsZmodnZMatrixRep" Arg='obj' Type='Representation'/>
+##
+##  <Description>
+##  An object <A>obj</A> in <Ref Filt="IsZmodnZMatrixRep"/> describes
+##  a matrix object (see <Ref Filt="IsMatrixObj"/>) that behaves like the
+##  list of its rows (see <Ref Filt="IsRowListMatrix"/>).
+##  <A>obj</A> is internally represented as a positional object
+##  (see <Ref Filt="IsPositionalObjectRep"/>) with <M>4</M> entries.
```

Comment:
As we just discussed on gather.town, I'd prefer to either keep these implementation specifics out of the user-visible documentation, or else to at least qualify them to indicate that these are subject to change and one should not write code exploiting them but rather use official interfaces.